rclcpp/test/rclcpp/test_subscription_options.cpp

```diff
@@ -54,6 +54,16 @@ class TestSubscriptionOptions : public ::testing::Test
   rclcpp::Node::SharedPtr node;
 };
 
+TEST_F(TestSubscriptionOptions, subscription_options_constructor) {
+  rclcpp::SubscriptionOptions option1;
+  EXPECT_EQ(option1.callback_group, nullptr);
+
+  auto group = std::make_shared<rclcpp::CallbackGroup>(rclcpp::CallbackGroupType::Reentrant);
+  rclcpp::SubscriptionOptions option2;
+  option2.set_callback_group(group);
+  EXPECT_EQ(option2.callback_group, group);
```

Comment:
```suggestion
  auto group = std::make_shared<rclcpp::CallbackGroup>(rclcpp::CallbackGroupType::Reentrant);
  auto option = rclcpp::SubscriptionOptions().set_callback_group(group);
  EXPECT_EQ(option.callback_group, group);
```

I deleted `option1`, as it didn't see to be adding much to the test.
I've also made a one line example of how the method is useful, though as commented above it would be better to fully enable method chaining in this PR.

Reply:
Thank you for reviewing.
I wanted comments before writing a lot.

> It would be gread to add methods enabling chaining of all options.

I'll be glad to do it!